### DISCUSSION TITLE
In example ep_slcoro_mqtts_client, ctx_ modifications are not used

### DISCUSSION THREAD

**Discussion by mcsmit** [redboltz/async_mqtt#360] [Q&A] [answered]
For my purpose, I modified the ep_slcoro_mqtts_client example to validate the server certificate and send a client certificate chain. I made my modifications in the app's constructor and changed `ctx_.set_verify_mode(as::ssl::verify_none)` to `ctx_.set_verify_mode(as::ssl::verify_none)`, followed by adding a client certificate chain and the private key. On purpose, I did not set a verify file to make sure my changes were picked up but were of no result.
I got the impression that in the app's constructor ctx_ is copied somehow and the original ctx_ is no longer used. To verify this, I added an extra param in the app's constructor, created the ctx in the main function like below and then my changes were picked up. Am I right that ctx_ is copied in the app's constructor?

```
int main(int argc, char* argv[]) {
    if (argc != 3) {
        std::cout << "Usage: " << argv[0] << " host port" << std::endl;
        return -1;
    }
    am::setup_log(am::severity_level::trace);
    as::io_context ioc;
    as::ssl::context ctx{as::ssl::context::tlsv13};
    ctx.set_verify_mode(as::ssl::verify_peer);
    ctx.load_verify_file("CA.crt");
    ctx.use_certificate_chain_file("chain.pem");
    ctx.use_private_key_file("certificate.key", as::ssl::context_base::pem);
    app a{ioc.get_executor(), argv[1], argv[2], ctx};
    ioc.run();
}
```

**Comment by redboltz** [comment]
`async_mqtt::basic_endpoint` forward args to the next layer. So https://www.boost.org/doc/libs/1_86_0/doc/html/boost_asio/reference/ssl__stream/stream/overload1.html is finally called.
The parameter is reference.

However, it is not equal to the setting is refelected after passing.

Let me clarify what you mean

https://github.com/redboltz/async_mqtt/blob/main/example/ep_slcoro_mqtts_client.cpp#L20-L33

```cpp
template <typename Executor>
struct app {
    app(Executor exe,
        std::string_view host,
        std::string_view port
    ):host_{std::move(host)},
      port_{std::move(port)},
      amep_{
          am::protocol_version::v3_1_1, exe, ctx_ // passed ctx_
      }
    {
        ctx_.set_verify_mode(as::ssl::verify_none); // set ctx_ **after** passed, but it is not reflected to the behavior
        impl_();
    }
```

I added comments. Is it correct ?

By the wey, please use issue https://github.com/redboltz/async_mqtt/issues instead of discusstions next time.

**Reply by redboltz** [reply, marked as answer]
I guess I understand what you mean correctly. If it does, the example is misleading, so I wrote the PR #361 
See `ep_slcoro_mqtts_client.cpp` part.

**Reply by mcsmit** [reply]
Yes, this is exactly what I discovered.
Next time, I will use issue.
